File: built-in plugins dsp/mb-drums-jazz.cpp
/**
 * MB Jazz Kit
 * Category : instrument
 * Type     : drums
 * Version  : 1.0.0
 * Author   : Max Booster
 * Desc     : Smooth jazz brush kit
 *
 * Part of Max Booster Built-In Plugins DSP
 */

#ifndef MB_DRUMS_JAZZ_H
#define MB_DRUMS_JAZZ_H

#include <algorithm>
#include <cmath>
#include <cstring>
#include "PluginBase.h"

class MbDrumsJazz : public PluginBase {
public:
    static constexpr const char* PLUGIN_ID      = "mb-drums-jazz";
    static constexpr const char* PLUGIN_NAME    = "MB Jazz Kit";
    static constexpr const char* PLUGIN_TYPE    = "drums";
    static constexpr const char* PLUGIN_CATEGORY = "instrument";
    static constexpr const char* VERSION         = "1.0.0";

    struct Parameters {
    float brush = 0.6f;  // range [0, 1]
    float volume = 0.7f;  // range [0, 1]
    };

    MbDrumsJazz() = default;
    ~MbDrumsJazz() override = default;

    void setSampleRate(double sampleRate) override {
        sampleRate_ = sampleRate;
        reset();
    }

    void reset() override {
        std::memset(buffer_, 0, sizeof(buffer_));
    }

    void process(float* left, float* right, int numSamples, Parameters params) {
        params.brush = std::clamp(params.brush, 0f, 1f);
        params.volume = std::clamp(params.volume, 0f, 1f);
        for (int i = 0; i < numSamples; ++i) {
            left[i]  = processSample(left[i],  params);
            right[i] = processSample(right[i], params);
        }
    }

private:
    double sampleRate_ = 44100.0;
    float  buffer_[65536] = {};

    inline float processSample(float input, const Parameters& params) {
        // DSP implementation for MB Jazz Kit
        return input;
    }
};

#endif // MB_DRUMS_JAZZ_H

File: built-in plugins dsp/mb-drums-jazz.ts
import type { PluginDefinition } from '../server/services/pluginHostService';

const MbDrumsJazzPlugin: PluginDefinition = {
    id: 'mb-drums-jazz', slug: 'mb-drums-jazz', name: 'MB Jazz Kit', category: 'instrument', type: 'drums', version: '1.0.0',
    description: 'Smooth jazz brush kit', author: 'Max Booster',
    oscillators: [{ type: 'triangle', detune: 0, gain: 0.6 }, { type: 'noise', detune: 0, gain: 0.4 }],
    envelope: { attack: 0.005, decay: 0.3, sustain: 0.1, release: 0.4 },
    parameters: [
      { id: 'brush', name: 'Brush', type: 'float', defaultValue: 0.6, minValue: 0, maxValue: 1, automatable: true },
      { id: 'volume', name: 'Volume', type: 'float', defaultValue: 0.7, minValue: 0, maxValue: 1, automatable: true },
    ],
    defaultPreset: { brush: 0.6, volume: 0.7 },
  };

export default MbDrumsJazzPlugin;
